server/src/main/java/io/crate/expression/reference/sys/node/NodeStatsContext.java

```diff
@@ -78,10 +80,43 @@ public class NodeStatsContext implements Writeable {
     private Integer httpPort;
     private Integer pgPort;
 
-    public NodeStatsContext(String id, String name) {
+
+    public enum Role {
```

Comment:
What about moving this into the `DiscoveryNode`?

Reply:
I think we should at least ensure the info isn't scattered around. If we introduce a node role concept, it shouldn't be isolated in the sys.node context but part of `DiscoveryNode`. E.g. have a `DiscoveryNode.getRoles()` and could also look into using a sealed approach for `DiscoveryNodeRole`, similar to how it is used in `RelationMetadata`.

I'm also not sure if we want the `MASTER` part of it. What do you think of keeping the role as `master-eligible` and `data` but have elected-master be a separate info. That would also match how `DiscoveryNodeRole` is currently modelled.